packages/playwright-core/src/server/firefox/ffBrowser.ts

```diff
@@ -331,7 +331,7 @@ export class FFBrowserContext extends BrowserContext {
 
   async doSetHTTPCredentials(httpCredentials?: types.Credentials): Promise<void> {
     this._options.httpCredentials = httpCredentials;
-    await this._browser._connection.send('Browser.setHTTPCredentials', { browserContextId: this._browserContextId, credentials: httpCredentials || null });
+    await this._browser._connection.send('Browser.setHTTPCredentials', { browserContextId: this._browserContextId, credentials: httpCredentials && { username: httpCredentials.username, password: httpCredentials.password } || null });
```

Comment:
Could you elaborate on the semantics of this expression?

```js
httpCredentials && { username: httpCredentials.username, password: httpCredentials.password } || null
```

Reply:
My mistake. I'll fix that.